lib/logstash/outputs/google_cloud_storage.rb

```diff
@@ -199,6 +197,18 @@ def close
 
   private
 
+  def legacy_encode(event, encoded)
+    # use-codec uses whatever the codec output exactly
+    return encoded if @output_format == 'use-codec'
```

Comment:
minor optimization: we could set both
```
@codec_output = @output_format == 'use-codec'
@json_output = @output_format == 'json'
```
and use these below to avoid continuously doing string comparisons.